src/content/operations/nat-traversal/_index.en.md

```diff
@@ -0,0 +1,415 @@
+port of packets, but **this will happen as soon as two connections collide
+over the same source port**.
+{{% /notice %}}
+
+## UDP Dataplane protocol (IPSEC & wireguard)
+
+By default submariner uses the 4500 UDP port for the dataplane. The cluster-wide default
+can be changed with the `subctl join` `--nattport` flag.
+
+Although, it's more advisable to specify the port to be used per gateway node:
+
+```bash
+kubectl label node $NODE gateway.submariner.io/udp-port=4501
+```
+
+This allows individual gateways on the cluster to have different port numbers,
+hence allowing individual port-mapping if a public IP is shared.
+
+## IPsec
+
+### ESP or UDP Encapsulation
+
+IPsec in the libreswan cable driver will be configured for ESP protocol (more performant)
+whenever that is possible, that's normally when NAT is not detected, likely to happen
+when connectivity over the private IP is possible.
+
+If your network and routers filter the `IP>ESP` packets, encapsulation can be forced
+by using the `--force-udp-encaps` during subctl join, or by setting the operator `
+
+## Practical examples
+
+### All private and routed
+
+This is the simplest practical case, in this case all gateways can contact
+all other gateways via routing on their private IPs, and no NAT happens.
+
+<!-- source: https://app.diagrams.net/#G1IORJ9qW95qJ15P6Xk6idEfI4jj9Uo84l -->
+![All private and routed](/images/natt/nat_scenario_all_private_routed.svg)
+
+Two configuration alternatives exist:
+
+* Enabling the NATT discovery protocol on all gateway nodes:
+
+```bash
+kubectl label node $GWA --kubeconfig A gateway.submariner.io/natt-discovery-port=4490
+kubectl label node $GWB --kubeconfig B gateway.submariner.io/natt-discovery-port=4490
+kubectl label node $GWC --kubeconfig C gateway.submariner.io/natt-discovery-port=4490
+kubectl label node $GWD --kubeconfig D gateway.submariner.io/natt-discovery-port=4490
+
+# restart the gateways to pick up the new setting
+for cluster in A B C D;
+do
+  kubectl delete pod -n submariner-operator -l app=submariner-gateway --kubeconfig $cluster
+done
+```
+
+* Deploy the clusters with `subctl join --natt=false`
+
+{{% notice note %}}
+The first alternative is the preferred one, since it will allow the eventual
+addition of a cluster on a public cloud, or a remote site with no direct IP
+reachability or NATed reachability.
+{{% /notice %}}
+
+### All public cloud, with some private reachability
+
+In this case case, A & B cluster gateways have direct reachability over their private
+IPs (10.0.0.1 and 10.1.0.1) possibly with large MTU capabilities, happening the
+same for C & D cluster gateways (192.168.0.4 and 192.168.128.4).
+
+Between A & C , A & D, B & C, and B & D reachability is still possible over
+the public IP, the IP packets will undergo DNAT + SNAT translation at the
+border via the elastic or floating IP, and also while on transit via the public
+network the MTU will be limited to 1500 bytes or less.
```

Comment:
```suggestion
Reachability between the two pairs of clusters is only possible over
their public IPs and the IP packets will undergo DNAT + SNAT translation at the
border via the elastic or floating IP and also, while on transit via the public
network, the MTU will be limited to 1500 bytes or less.
```

Reply:
A & B and C&D reach over their private IPs so not all two pairs of clusters connect via public IPs.

Reply:
I rephrased some bits and applied.